packages/spanner/src/transaction.js

```diff
@@ -283,11 +314,28 @@ Transaction.prototype.commit = function(callback) {
  * @param {function} callback - The callback function.
  */
 Transaction.prototype.request = function(config, callback) {
+  var self = this;
+
   var reqOpts = extend({
     session: this.session.formattedName_
   }, config.reqOpts);
 
-  config.method(reqOpts, callback);
+  var gaxOptions = reqOpts.gaxOptions;
+  delete reqOpts.gaxOptions;
+
+  config.method(reqOpts, gaxOptions, function(err, resp) {
```

Comment:
@stephenplusplus so I refactored the way we handle aborted errors from transactions - however, I'm not 100% how I should handle this for streams.

My understanding is that reads can trigger an abort error, so my thought would be to create a new through stream and pipe the gax stream to it, handle errors in the same way as below and if we get an aborted error to destroy the stream?

My overall concern here is that abort can probably occur in between requests, meaning the user could get a handful of results before we have to retry the whole transaction.